### PR TITLE
Add cli.

### PR DESCRIPTION
**Pull request by dvc94ch**
Useful for debugging ipfs stuff. Also planning on adding a `multihash` and `cid` cli.

### CODE REVIEW

**Comment by dvc94ch** [comment]
base_x takes 500us to encode and 80us to decode base32 and base64.

base32 takes 3us and base64 takes 700ns.

**Comment by dvc94ch** [comment]
Closes #11, #9 #10 and removes a bunch of implementations that are plain wrong. @dignifiedquire thoughts? :)

File: cli/main.rs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            Mode::Encode { base } => encode(base).await,
+            Mode::Decode => decode().await,
+        }
+    })
+}
+
+#[derive(Debug)]
+struct StrBase(Base);
+
+impl fmt::Display for StrBase {
```
Comment (dignifiedquire) [inline cli/main.rs]: maybe use [strum](https://github.com/Peternator7/strum) to avoid all the repetition below?

File: cli/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+[package]
+name = "multibase-cli"
+version = "0.1.0"
+authors = ["David Craven <david@craven.ch>"]
+edition = "2018"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+async-std = "0.99.4"
```
Comment (dignifiedquire) [inline cli/Cargo.toml]: 10/10 for using async-std :)

Comment (dignifiedquire) [reply cli/Cargo.toml]: but on the other hand I am wondering, do we actually need it, given that there is very little that the cli actually does

**Comment by dvc94ch** [comment]
go-multibase is rfc4648 compliant, so they should match now. The tests I added are straight from the spec...

**Comment by dignifiedquire** [comment]
Thank you, will publish soon

**Pull request merged**